types/video.js/index.d.ts

```diff
@@ -2054,6 +2054,20 @@ declare namespace videojs {
          * @return The element that was created.
          */
         createEl(): HTMLDivElement;
+
+        /**
+         * Returns the child `Component` with the given `name`.
+         *
+         * @param name
+         *        The name of the child `Component` to get.
+         *
+         * @return The child `Component` with the given `name` or undefined.
+         */
+        getChild(name: string): ControlBarChild | undefined;
+    }
+
+    interface ControlBarChild extends Component {
+        el(): HTMLDivElement;
```

Comment:
It's perfectly valid for a component to be added to ControlBar that is not a div so we can't say that all ControlBarChildren#el() calls will be divs. Component#el() already returns Element so this interface is not needed. Same really for the getChild method you're overriding. Component#getChild(string) returns Component already which has el() that returns Element. 

However, you can add more specific overloads:

```ts
interface ControlBar {
        // your overload
        getChild(name: 'playbackRateMenuButton'): PlaybackRateMenuButton | undefined;
        // existing
        getChild(name: string): Component | undefined;
}
```

But realistically it should probably still be done in Component#getChild and use a map rather than typing out each overload of getChild

```ts
// subclass Component and override el() for those where we know which specific type of element is returned
interface PlaybackRateMenuButton extends Component {
  el(): HTMLDivElement;
}

// create a map of names to components
type ComponentNameMap {
  playbackRateMenuButton: PlaybackRateMenuButton,
  ...otherComponent: OtherComponent
}

interface Component {
        // read specific component types from our map
        getChild<TComponentName extends keyof ComponentNameMap>(name: TComponentName): ComponentNameMap[TComponentName] | undefined;
        // fall back to returning generic Component
        getChild(name: string): Component | undefined;
}
```

I didn't test the above but something along those lines would work

Reply:
Oh... I though el() would always return a DivElement, because createEl() returns a DivElement.
well, I added the ComponentNameMap as you suggested :)